apigee_sync_test.go

```diff
@@ -155,7 +155,7 @@ var _ = Describe("Sync", func() {
 			}
 			pie.Plugins = append(pie.Plugins, pluginData)
 			postInitPlugins(pie)
-		}, 3)
+		}, 5)
```

Comment:
This is because I added graceful shutdown of token thread, change thread, and snapshot thread. I use blocking shutdown to ensure nothing is left after the shutdown and no output of `-race`. The bootstrap in `postInitPlugins` also took some time, which was counted into the timeout. 
Most of the problems in the old ApigeeSync were caused by ungraceful shutdown. The threads left will mess up with latter tests.
The tests are much faster locally. But the CI may sometimes take more than 3 secs.